builder/hcloud/step_create_server.go

```diff
@@ -142,6 +143,16 @@ func (s *stepCreateServer) Run(ctx context.Context, state multistep.StateBag) mu
 	}
 	state.Put(StateServerIP, serverIP)
 
+	// Wait that the server to settle before continuing. Prevents possible `locked`
+	// error when changing the server type.
+	actions, err := getServerRunningActions(ctx, client, server)
+	if err != nil {
+		return errorHandler(state, ui, "Could not fetch server running actions", err)
+	}
+	if err := client.Action.WaitFor(ctx, actions...); err != nil {
+		return errorHandler(state, ui, "Could not wait for server running actions", err)
+	}
```

Comment:
As this is a nice-2-have, we could also log a warning and just continue here.

Reply:
I am not sure if it is "nice to have", I'd rather say it required for the use cases we propose: making sure a firewall is applied before running further actions on the server.